src/contextbuilder.cpp

```diff
@@ -84,6 +84,62 @@ std::vector<std::string> ContextBuilder::listDevices()
 	return uris;
 }
 
+std::string ContextBuilder::getUri(struct iio_context* ctx)
+{
+	if(ctx == nullptr)
+        {
+		throw_exception(EXC_INVALID_PARAMETER, "Invalid iio context");
+        }
+	std::string uri;
+	std::string context_type(iio_context_get_name(ctx));
+
+	if(context_type == "network")
```

Comment:
for a getter, this looks like it's doing a lot;
well, a lot of stuff that libiio is doing;

maybe try to create a context object with libiio, and if that works, return the URI as string
usually, and URI is passed on from a user, so we should have the original/complete URI already available somewhere

Reply:
You are right. The problem was fixed in commit 5f6c11f.